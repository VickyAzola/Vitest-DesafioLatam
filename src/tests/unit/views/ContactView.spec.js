import { describe, it, expect, test } from 'vitest'
import { mount } from '@vue/test-utils'
import { createRouter, createWebHistory } from 'vue-router'
import ContactView from '@/views/ContactView.vue'

describe('ContactView', () => { 
    const wrapper = mount(ContactView)

    it('El HTML del componente se mantiene durante las pruebas', () => {
        expect(wrapper.html()).toMatchSnapshot()
    })

    test('Probando la existencia de la vista ContactView', async () => {
        const router = createRouter({
            history: createWebHistory(),
            routes: [{
                path: '/contact',
                name: 'contact',
                component: ContactView
            }],
        })

        router.push({name: 'contact'})
        
        await router.isReady()

        const wrapper = mount(ContactView, {
            global: {
                plugins: [router]
            }
        })

        expect(wrapper.findComponent(ContactView).exists()).toBe(true)
    })  
})