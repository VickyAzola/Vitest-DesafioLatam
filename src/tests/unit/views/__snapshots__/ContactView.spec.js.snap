// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`ContactView > El HTML del componente se mantiene durante las pruebas 1`] = `"<h1>Contact View</h1>"`;
